portfolio/src/main/java/com/google/sps/servlets/DeleterServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that will delete all comments. */
+@WebServlet("/delete-comments")
+public class DeleterServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Key> keysToDelete = new ArrayList<>();
+    for (final Entity entity : results.asIterable()) {
+      final long id = entity.getKey().getId();
+      keysToDelete.add(KeyFactory.createKey("Comment", id));
+    }
+    for (final Key keyToDelete : keysToDelete) {
```

Comment:
I think we don't need this for-loop. Can we put `datastore.delete(keyToDelete); ` into the previous for-loop?

Reply:
I had this for loop because I was concerned that there would be issues with modifying the db while iterating over the results, but I tried changing it and it appears that that did not introduce issues.

Reply:
If I'm looking at the correct documentation, delete supports taking a list of keys (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService?hl=en#method-summary). So we can directly pass the list to reduce the traffic to db.